Sources/ProjectDescription/Template.swift

```diff
@@ -0,0 +1,76 @@
+public struct Template: Codable {
+    public let description: String
+    public let attributes: [Attribute]
+    public let files: [File]
+    public let directories: [String]
+    
+    public init(description: String,
+                arguments: [Attribute] = [],
+                files: [File] = [],
+                directories: [String] = []) {
+        self.description = description
+        self.attributes = arguments
+        self.files = files
+        self.directories = directories
+        dumpIfNeeded(self)
+    }
+}
+
+public extension String.StringInterpolation {
+    mutating func appendInterpolation(_ value: Attribute) {
+        switch value {
+        case let .required(name), let .optional(name, default: _):
+            appendInterpolation("{{ \(name) }}")
+        }
+    }
+}
+
+public struct File: Codable {
```

Comment:
Similar comment here in regards to placing the definition of `File` under `Template`.

Reply:
Yes, I will fill in the documentation once we are settled on the `template` manifest.